modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -308,15 +308,30 @@ GrainTracker::trackGrains()
         mooseError("Error with ESBD Mapping (see above unused indices)");
       }
     }
-    else // Just assign IDs in order
+    else
     {
+      /**
+       * Here we want to assign the grains in some partitioning invariant way. We'll sort first by
+       * _var_idx (already partitioned in that manner) then sort on the _min_entity_id
+       * to order the grains.
+       */
       unsigned int counter = 0;
       for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
```

Comment:
Fine with me.

On Thu, Jun 9, 2016 at 12:58 PM Derek Gaston <notifications@github.com>
wrote:

> In modules/phase_field/src/postprocessors/GrainTracker.C
> <https://github.com/idaholab/moose/pull/7161#discussion_r66501265>:
>
> >      {
> > +      /**
> > +       * Here we want to assign the grains in some partitioning invariant way. We'll sort first by
> > +       * _var_idx (already partitioned in that manner) then sort on the _min_entity_id
> > +       * to order the grains.
> > +       */
> >        unsigned int counter = 0;
> >        for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
>
> Interesting. I really don't like auto map_num = decltype(_maps_size)(0).
> It goes against our guidelines that say not to use constructor () syntax
> for simple objects! And if this wasn't a simple object then it would go
> against our guidelines to not use assignment construction syntax for
> complicated objects.
>
> So, it's a double whammy on our style guidelines.
>
> decltype(_maps_size) map_num = 0 is very easy to parse. In my mind
> decltype() is just like a preprocessor directive / macro... and my mind
> very happily just swaps in the type of _maps_type right into the code
> there so it looks like a normal for loop.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7161/files/42f650de3fa3cd81c18827c9144c1d2ee08bd5af#r66501265>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC5XIH2BeZA76rhsMTKkc6BwHkHt2yXqks5qKGJtgaJpZM4IwRk6>
> .
>